tests/test_dynamodb2/test_dynamodb.py

```diff
@@ -1641,6 +1718,89 @@ def test_condition_expressions():
             }
         )
 
+    # Make sure update_item honors ConditionExpression as well
+    client.update_item(
+        TableName='test1',
+        Key={
+            'client': {'S': 'client1'},
+            'app': {'S': 'app1'},
+        },
+        UpdateExpression='set #match=:match',
+        ConditionExpression='attribute_exists(#existing)',
+        ExpressionAttributeNames={
+            '#existing': 'existing',
+            '#match': 'match',
+        },
+        ExpressionAttributeValues={
+            ':match': {'S': 'match'}
+        }
+    )
+
+    with assert_raises(client.exceptions.ConditionalCheckFailedException):
+        client.update_item(
+            TableName='test1',
+            Key={
+                'client': { 'S': 'client1'},
+                'app': { 'S': 'app1'},
+            },
+            UpdateExpression='set #match=:match',
+            ConditionExpression='attribute_not_exists(#existing)',
+            ExpressionAttributeValues={
+                ':match': {'S': 'match'}
+            },
+            ExpressionAttributeNames={
+                '#existing': 'existing',
+                '#match': 'match',
+            },
+        )
+
+
+@mock_dynamodb2
+def test_condition_expression__attr_doesnt_exist():
+    client = boto3.client('dynamodb', region_name='us-east-1')
+
+    client.create_table(
+        TableName='test',
+        KeySchema=[{'AttributeName': 'forum_name', 'KeyType': 'HASH'}],
+        AttributeDefinitions=[
+            {'AttributeName': 'forum_name', 'AttributeType': 'S'},
+        ],
+        ProvisionedThroughput={'ReadCapacityUnits': 1, 'WriteCapacityUnits': 1},
+    )
+
+    client.put_item(
+        TableName='test',
+        Item={
+            'forum_name': {'S': 'foo'},
+            'ttl': {'N': 'bar'},
+        }
+    )
+
+
+    def update_if_attr_doesnt_exist():
```

Comment:
@killthrush I think this captures your comment from #2136 but let me know if not!

Reply:
Thanks for picking this up @garrettheel!  This fixes one issue, however in the interim I've collected a few more test cases related to conditional checks that still fail with this commit.  I'll see if I can capture them too by cloning your branch.

Reply:
Hmm, it looks like some of these failures are unrelated to moto.  I'll keep testing, but feel free to move ahead with this PR - I can always file an issue after the fact.  Thank you!

Reply:
OK, I *did* find some test failures related to `list_append` not working properly, but I suspect that is a different issue judging from the commit history.